### PR TITLE
Website model/API changes needed for phase 0 (bug 1162325)

### PR DESCRIPTION
**Pull request by diox**
https://bugzilla.mozilla.org/show_bug.cgi?id=1162325

- `name` is introduced, it's the main field to use to display a website.
- `short_name` replaces `short_title`. 
- `title` is kept, but is not really meant to be exposed to the consumer pages. It's helpful to increase a website relevancy though.
- `mobile_url` is added for websites that have a mobile-specific version at a different URL.

### CODE REVIEW

File: mkt/websites/tests/test_views.py [hunk 1]
```diff
@@ -53,10 +53,12 @@ def test_basic(self):
...
         eq_(data['description'], {'en-US': self.website.description})
         eq_(data['title'], {'en-US': self.website.title})
-        eq_(data['short_title'], {'en-US': self.website.short_title})
-        eq_(data['url'], {'en-US': self.website.url})
+        eq_(data['name'], {'en-US': self.website.name})
+        eq_(data['short_name'], {'en-US': self.website.short_name})
+        eq_(data['url'], self.website.url)
         eq_(data['device_types'], ['firefoxos', 'desktop'])
         eq_(data['categories'], ['books', 'sports'])
+        # FIXME: regions, keywords, icon
```
Comment (diox) [inline mkt/websites/tests/test_views.py]: icons is https://bugzilla.mozilla.org/show_bug.cgi?id=1164053 btw

File: mkt/search/tests/test_views.py [hunk 2]
```diff
@@ -1271,6 +1271,8 @@ def setUp(self):
         self.webapp.update(categories=[self.shared_category, 'business'])
         self.webapp.popularity.create(region=0, value=11.0)
         self.website = website_factory(
+            name='something something webcube',
```
Comment (washort) [inline mkt/search/tests/test_views.py]: webcube :+1:

**Comment by washort** [comment]
r+

File: mkt/search/tests/test_views.py (hunk 2 above)
Comment (robhudson) [inline mkt/search/tests/test_views.py]: this app lost all its steam :steam_locomotive: :dash:

**Pull request merged**